src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -91,6 +120,38 @@ export const Configurator = ({
     fetchFields();
   }, [baseDataset, indexPatternsService, queryString, timeFields.length]);
 
+  const updateDatasetForIndexedView = useCallback(async () => {
+    if (!indexedViewsService || !selectedIndexedView) {
+      return dataset;
+    }
+
+    let connectedDataSource;
```

Comment:
can be a fast follow but i think the indexed view service could handle fetching the datasource and creating and returning an indexed view vs the configurator doing this work